x/bridge/types/msgs.go

```diff
@@ -0,0 +1,148 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgInboundTransfer{}
+
+func NewMsgInboundTransfer(
+	sender string,
+	destAddr string,
+	asset Asset,
+	amount math.Int,
+) *MsgInboundTransfer {
+	return &MsgInboundTransfer{
+		Sender:   sender,
+		DestAddr: destAddr,
+		Asset:    asset,
+		Amount:   amount,
+	}
+}
+
+func (m MsgInboundTransfer) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(m.Sender)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+	}
+
+	_, err = sdk.AccAddressFromBech32(m.DestAddr)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid destination address (%s)", err)
+	}
+
+	err = m.Asset.Validate()
+	if err != nil {
+		return errorsmod.Wrapf(ErrInvalidAsset, err.Error())
+	}
+
+	if !m.Amount.IsZero() {
+		return errorsmod.Wrap(sdkerrors.ErrInvalidCoins, m.Amount.String())
+	}
```

Comment:
Should we just be extra certain and careful by adding a negative check here as well (also in outbound)?

Reply:
Done